circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -76,14 +76,33 @@ trait CirceInstances extends JawnInstances {
   def jsonOf[F[_]: Sync, A: Decoder]: EntityDecoder[F, A] =
     jsonOfWithMedia(MediaType.application.json)
 
+  def jsonOfSensitive[F[_]: Sync, A: Decoder]: EntityDecoder[F, A] =
+    jsonOfWithSensitiveMedia(MediaType.application.json)
+
   def jsonOfWithMedia[F[_], A](r1: MediaRange, rs: MediaRange*)(implicit
       F: Sync[F],
       decoder: Decoder[A]): EntityDecoder[F, A] =
+    jsonOfWithMediaHelper[F, A](r1, jsonDecodeError, rs: _*)
+
+  def jsonOfWithSensitiveMedia[F[_], A](r1: MediaRange, rs: MediaRange*)(implicit
+      F: Sync[F],
+      decoder: Decoder[A]): EntityDecoder[F, A] =
+    jsonOfWithMediaHelper[F, A](
+      r1,
+      (json, nelDecodeFailures) =>
+        CirceInstances.jsonDecodeErrorHelper(json, _ => "<REDACTED>", nelDecodeFailures),
```

Comment:
The `String => String` sanitization is a fair concern on the `SanitizedDecodeFailure`.  But we're just doing a constant function here.  To take advantage of having a `Json`, would it make sense to lift this function up as an argument, so we could sanitize the sensitive fields and still be able to debug the rest?

A counterargument to that is we're here due to an invalid JSON structure, and the sensitive bits might be under a JSON key with a typo.

Reply:
> To take advantage of having a Json, would it make sense to lift this function up as an argument, so we could sanitize the sensitive fields and still be able to debug the rest?

If I understand properly, you're proposing re-writing `CirceSensitiveDataEntityDecoder` as:

```scala
trait CirceSensitiveDataEntityDecoder {
  def circeEntityDecoder[F[_]: Sync, A: Decoder](sanitize: Json => String): EntityDecoder[F, A] =
    jsonOfSensitive[F, A]
}
```

, @rossabaker?

I'm not convinced that's your suggestion since we'd lose the `implicit`.

> A counterargument to that is we're here due to an invalid JSON structure, and the sensitive bits might be under a JSON key with a typo.

Yep - good point.

Reply:
I was proposing parameterizing `jsonOfWithSensitiveMedia` with that function, and fixing up the compile errors that bubble up.  You could use your `_ => "<REDACTED>"` as a default.  I might be underestimating all the places that ripples.

Reply:
Hey @rossabaker - I like that idea of passing an argument of `Json => String`. 

Can you please review https://github.com/http4s/http4s/pull/4012#commits-pushed-cba03fe?

I opted against the default parameter, but, please let me know what you prefer.